config/api/routes/users.php

```diff
@@ -203,7 +203,9 @@ function ($source, $filename) use ($id) {
 			'users/(:any)/roles',
 		],
 		'action'  => function (string $id) {
-			return $this->user($id)->roles();
+			$kirby   = $this->kirby();
+			$purpose = $kirby->request()->get('purpose');
+			return $this->user($id)->roles($purpose);
```

Comment:
I would prefer such a simple usage instead of making the `User::roles()` method more complicated.

```suggestion
            $roles = $this->user($id)->roles();

            return match ($purpose) {
			    'create' => $roles->canBeCreated(),
			    'change' => $roles->canBeChanged(),
			    default  => $roles
			};
```

Reply:
I think I agree with @afbora. Alternatively, I would probably vote to have two methods with a single purpose each, instead of the additional argument.

Reply:
But we will need this in a couple of more places where we have so far forgotten to apply the filters. I think we should do it in one central place otherwise the danger is that we keep forgetting it when we change something

Reply:
Understood. Then it makes more sense to apply Bastian's idea of ​​a secondary separated method, for example: `->canBe($action)`

Reply:
But we need it on the user object, not the roles object.

Reply:
Sorry, my mistake. `User::canBe()` or something like that.

Reply:
I was really thinking about two individual methods. Something like User::canCreateRoles and User::canChangeRoles but after trying to find some really good method names, I'm wondering if we are overthinking it and the argument really is the more practical option. Naming is hard :)

Reply:
I think the argument is for me the one to not as easily forget about. If I know `->roles()`, I will spot the optional argument via the IDE when using it. Remembering that there are also `->canCreateRoles()` etc. might be harder.

Reply:
If you feel very strongly about not touching `->roles()`, I think the alternative method should be named very closely, e.g. `->rolesByPurpose($purpose)`, `->rolesFor($purpose)`, `->rolesTo($action)` ...